src/rgw/cls_fifo_legacy.cc

```diff
@@ -404,7 +404,7 @@ std::optional<marker> FIFO::to_marker(std::string_view s)
   }
 
   auto pos = s.find(':');
-  if (pos == string::npos) {
+  if (pos == s.npos) {
```

Comment:
the 'before' is the idiomatic way, AFAIK.
(see for example https://en.cppreference.com/w/cpp/string/basic_string/npos)

Reply:
@ronen-fr i'd like to understand why `string::npos` is better than `s.npos` apart from that cppreference uses the former. i admit that i cannot find enough proof to prove that it the idiomatic way or disprove it, cppreference is not enough to prove this, IMHO. i agree that, it's a good source as a human-readable version of C++ standards. but if you could help me understand 'before' is *better*, i'd be more than happy change it.

Reply:
'npos' is static, and that's one good reason to use it as string::npos. It's not a property of the specific string.
Also - it's not just cppref. That's the way you usually find it: in Microsoft docs, or (by chance) in the following
book: "Programming and Problem Solving with C++", and (judging by a google search) - "everywhere".

Don't have time now to look for the relevant standard doc.

Reply:
one advantage of `s.npos` is that it's generic to the underlying string type - which, in this case, is a `std::string_view`, so `string::npos` wasn't actually correct